diff --git a/config.layout b/config.layout
index da63e89..d46bfaf 100644
--- a/config.layout
+++ b/config.layout
@@ -219,6 +219,27 @@
     cachedir:      /var/cache+
 </Layout>
 
+#   openEuler Linux TrafficServer path layout.
+<Layout openEuler>
+    prefix:        /usr
+    exec_prefix:   ${prefix}
+    bindir:        ${exec_prefix}/bin
+    sbindir:       ${exec_prefix}/sbin
+    libdir:        ${exec_prefix}/lib64+
+    libexecdir:    ${libdir}/plugins
+    infodir:       ${prefix}/share/info
+    mandir:        ${prefix}/share/man
+    sysconfdir:    ${prefix}/etc+
+    datadir:       ${prefix}/share+
+    docdir:        ${prefix}/share/doc+
+    installbuilddir: ${datadir}/build
+    includedir:    ${prefix}/include
+    localstatedir: ${prefix}/var+
+    runtimedir:    ${localstatedir}/run
+    logdir:        ${localstatedir}/log
+    cachedir:      ${localstatedir}/cache
+</Layout>
+
 #   OpenBSD path layout.
 <Layout OpenBSD>
     prefix:        /usr/local
-- 
2.27.0

